spec/forms/candidate_interface/restructured_work_history/job_form_spec.rb

```diff
@@ -0,0 +1,202 @@
+      commitment: %w[full_time part_time].sample,
+      start_date: Time.zone.local(2018, 5, 1),
+      start_date_unknown: false,
+      end_date: Time.zone.local(2019, 5, 1),
+      end_date_unknown: false,
+      currently_working: false,
+      relevant_skills: [true, false].sample,
+    }
+  end
+
+  let(:form_data) do
+    {
+      role: data[:role],
+      organisation: data[:organisation],
+      commitment: data[:commitment],
+      start_date_month: data[:start_date].month,
+      start_date_year: data[:start_date].year,
+      start_date_unknown: data[:start_date_unknown],
+      end_date_month: data[:end_date].month,
+      end_date_year: data[:end_date].year,
+      end_date_unknown: data[:end_date_unknown],
+      currently_working: data[:currently_working],
+      relevant_skills: data[:relevant_skills],
+    }
+  end
+
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:role) }
+    it { is_expected.to validate_presence_of(:organisation) }
+    it { is_expected.to validate_presence_of(:commitment) }
+
+    it { is_expected.to validate_length_of(:role).is_at_most(60) }
+    it { is_expected.to validate_length_of(:organisation).is_at_most(60) }
+
+    include_examples 'validation for a start date', 'restructured_work_history/job_form'
+
+    context "'currently working' is 'false'" do
+      it 'is invalid if left completely blank' do
+        form = described_class.new(end_date_month: '', end_date_year: '', currently_working: false)
+
+        form.validate
+
+        expect(form.errors.full_messages_for(:end_date)).to eq(
+          ["End date #{t('activemodel.errors.models.candidate_interface/restructured_work_history/job_form.attributes.end_date.invalid')}"],
+        )
+      end
+
+      it 'is invalid if month left blank' do
+        form = described_class.new(end_date_month: '', end_date_year: '2019', currently_working: false)
+
+        form.validate
+
+        expect(form.errors.full_messages_for(:end_date)).to eq(
+          ["End date #{t('activemodel.errors.models.candidate_interface/restructured_work_history/job_form.attributes.end_date.invalid')}"],
+        )
+      end
+
+      it 'is invalid if year left blank' do
+        form = described_class.new(end_date_month: '5', end_date_year: '', currently_working: false)
+
+        form.validate
+
+        expect(form.errors.full_messages_for(:end_date)).to eq(
+          ["End date #{t('activemodel.errors.models.candidate_interface/restructured_work_history/job_form.attributes.end_date.invalid')}"],
+        )
+      end
+
+      it 'is invalid if not well-formed' do
+        form = described_class.new(end_date_month: '99', end_date_year: '2019', currently_working: false)
+
+        form.validate
+
+        expect(form.errors.full_messages_for(:end_date)).to eq(
+          ["End date #{t('activemodel.errors.models.candidate_interface/restructured_work_history/job_form.attributes.end_date.invalid')}"],
+        )
+      end
```

Comment:
Really thorough testing of the validations Toby 🙌 

I don't think it's a blocker for this, but we're now calling the methods from the `DateValidationHelper` in a few places in the application.

I wonder if we should really just be poritng the tests we have into a `date_validation_helper_spec.rb` file. At the moment i think we're calling it in a few places.

Also this exists 

![image](https://user-images.githubusercontent.com/42515961/106363934-7998e300-6323-11eb-8a7e01c227cb596f.png)

I'll add a tech debt card though. No need to block this.

Reply:
Yeah I was not able to add `include_examples 'validation for an end date'` because in the new flow we only validate the end date if the user selects that they are no longer working in that role. Given this is a unique case I did not think it would be appropriate to update the includes specs to reflect this (e.g. via a flag)